src/sentry/snuba/discover.py

```diff
@@ -0,0 +1,230 @@
+from __future__ import absolute_import
+
+import six
+
+from copy import deepcopy
+from sentry.api.event_search import get_filter, resolve_field_list
+from sentry.utils.snuba import (
+    Dataset,
+    DISCOVER_COLUMN_MAP,
+    QUOTED_LITERAL_RE,
+    get_function_index,
+    raw_query,
+)
+
+ALLOWED_INTERNALS = set(["project_id", "event_id"])
+
+
+def resolve_column(col):
+    """
+    Resolve a public schema name to the discover dataset.
+    unknown columns are converted into tags expressions.
+    """
+    if not col:
+        return ""
```

Comment:
When would col be None and why would we resolve that to `""` (which I assume will error on clickhouse). Would it be better to just raise the exception here

Reply:
This scenario comes up when traversing aggregates like `["count", "" "count"]`. I can make the check more specific to `""`.

Reply:
Would using the `["count", [], "count"]` or `["count()", None, "count"]` syntax help (I haven't tried yet so needs to be confirmed) avoid these problems? We are essentially making `""` a valid column name everywhere (which it shouldn't be) so we can abuse this syntax that only supports calling a function with a single argument.
I know that we probably do this in some other places already, but if we have some other syntax that supports none / multiple columns instead - let's try to use that?